Cpp/Library/Control/ReturnBuffer.hh

```diff
@@ -8,55 +8,118 @@
 #ifndef NYMPH_RETURNBUFFER_HH_
 #define NYMPH_RETURNBUFFER_HH_
 
+#include "Exception.hh"
+
 #include <tuple>
 
 namespace Nymph
 {
 
+    /*!
+     @class ReturnBufferBase
+     @author N. S. Oblath
+
+     @brief Base class for ReturnBuffers, providing access to the returned parameters
+
+     @details
+     ReturnBuffers are a type-erasure mechanism for returning parameters at a breakpoint in Nymph operations.
+
+     This base class provides the type erasure while grating access to the returned paramters via a templated GetREturn() function.
+    */
     class ReturnBufferBase
     {
         public:
             ReturnBufferBase();
             virtual ~ReturnBufferBase();
+
+            /// Provides access to the returned parameters
+            /// Throws Nymph::Exception if the buffer is empty or if XArgs... does not match the stored contents
+            template< class... XArgs >
+            std::tuple< XArgs&... >& GetReturn();
+
+            /// Provides const access to the returned parameters
+            /// Throws Nymph::Exception if the buffer is empty or if XArgs... does not match the stored contents
+            template< class... XArgs >
+            const std::tuple< XArgs&... >& GetReturn() const;
     };
 
-    template< typename... Args >
-    struct ReturnBuffer : ReturnBufferBase
+    /*!
+     @class ReturnBuffer<XArgs...>
+     @author N. S. Oblath
+
+     @brief Provides storage for returned parameters
+
+     @details
+     Parameters are stored as a `std::tuple<XArgs...>`.
+    */
+    template< class... XArgs >
+    class ReturnBuffer : public ReturnBufferBase
     {
         public:
-            ReturnBuffer() :
-                    fReturn( nullptr )
-            {}
-
-            ReturnBuffer( Args&... retval ) :
-                    fReturn( new std::tuple< Args&... >( retval... ) )
-            {}
+            ReturnBuffer();
+            ReturnBuffer( XArgs&... retval );
+            virtual ~ReturnBuffer();
 
-            virtual ~ReturnBuffer()
-            {}
+            /// Provides access to the returned parameters
+            /// Throws Nymph::Exception if the buffer is empty
+            std::tuple< XArgs&... >& GetReturn();
 
-            std::tuple< Args&... >& GetReturn()
-            {
-                if( fReturn == nullptr ) throw std::exception();
-                return *fReturn;
-            }
+            /// Provides const access to the returned parameters
+            /// Throws Nymph::Exception if the buffer is empty
+            const std::tuple< XArgs&... >& GetReturn() const;
 
         protected:
-            std::tuple< Args&... >* fReturn;
+            std::tuple< XArgs&... >* fReturn;
```

Comment:
This pointer raises a couple of questions regarding ownership. 
At first it looks like the pointer might not be necessary at all because `GetReturn` returns a reference anyway.